### PR TITLE
fix: Work around (temporary?) issues with the content book

### PR DESCRIPTION
**Pull request by CommandMC**

### CODE REVIEW

**Comment by CommandMC** [comment]
It seems this doesn't entirely resolve the issues. While scanning the content book works right after logging in now, it again breaks after opening it a few times

**Review by ryanzhoudev — APPROVED** [review]

**Comment by ryanzhoudev** [comment]
Discussion starts [here](https://discord.com/channels/394189072635133952/1257255724061425674/1295710267543519314) in case anyone wants to read it

File: common/src/main/java/com/wynntils/models/activities/ContentBookQueries.java [hunk 1]
```diff
@@ -245,19 +252,25 @@ private int getFilterChangeDirection(ItemStack itemStack, String targetFilter) {
...
+            // TEMP FIX: Consider the filter change complete as soon as we get the first item.
+            //           We don't seem to get called for changes in the player's inventory, so we cannot
+            //           rely on the Filter or Progress items changing.
+            //           We also can't wait for the last item, since we don't know which item that might be
+            //           (some filter categories have less than a page of items, so we won't always get
+            //           change events for the last item slot)
+            //           Additionally, it appears SET_CONTENT is now used in addition to SET_SLOT, but
+            //           unconfirmed if this is intentional or a temporary bug.
+            if (changeType != ContainerContentChangeType.SET_CONTENT
+                    && changeType != ContainerContentChangeType.SET_SLOT) return false;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/activities/ContentBookQueries.java:272]: What is the enum in this case? Null? I think this condition is either written wrongly by accident, or has a flaw in it's logic, or it masks a bigger issue in the query system (that being changeType being null).

**Review by kristofbolyai — COMMENTED** [review]
Let's wait for Zeer's response, we've asked him internally first.

File: common/src/main/java/com/wynntils/models/activities/ContentBookQueries.java [hunk 2]
```diff
@@ -245,19 +252,25 @@ private int getFilterChangeDirection(ItemStack itemStack, String targetFilter) {
...
-            if (!changes.containsKey(PROGRESS_SLOT)) return false;
-
-            // Check if the filter item changed, and if so, if the active filter changed
-            String itemFilter = getActiveFilter(container.items().get(CHANGE_VIEW_SLOT));
-            return !Objects.equals(itemFilter, activeFilter);
+            // if (changeType == ContainerContentChangeType.SET_CONTENT) return false;
+
+            // TEMP FIX: Consider the filter change complete as soon as we get the first item.
+            //           We don't seem to get called for changes in the player's inventory, so we cannot
+            //           rely on the Filter or Progress items changing.
```
Comment (CommandMC) [inline common/src/main/java/com/wynntils/models/activities/ContentBookQueries.java:265]: This part of the comment is now inaccurate

File: common/src/main/java/com/wynntils/models/activities/ContentBookQueries.java (hunk 1 above)
Comment (CommandMC) [reply common/src/main/java/com/wynntils/models/activities/ContentBookQueries.java:272]: Briefly testing here, I did not encounter a `changeType` deviating from the two enum values. I assume the condition is faulty & could be removed

**Comment by kristofbolyai** [comment]
Thank you for your PR, but due to how problematic this issue is on the whole mod, I've decided to handle the fix myself in a different PR. I'll close this now. Thank you for trying to help out.

**Comment by CommandMC** [comment]
That's totally fine, I assumed from the start that my solution was a little hacky, only really worked on this to give something to the people while a proper fix is developed

**Pull request closed without merging**